maps-db/src/main/resources/db/migration/V1.00.01__create_map_index_table.sql

```diff
@@ -0,0 +1,33 @@
+comment on database maps_db is 'Database used to store, track and serve map information';
+
+create table map (
+  id serial primary key,
+  name varchar(256) not null unique check(length(name) > 3),
+  date_created timestamptz not null default now(),
+  date_updated timestamptz not null default now()
+);
+
+create table category (
+  id serial primary key,
+  name varchar(32) unique not null check (length(name) > 2),
+  date_created timestamptz not null default now()
+);
+
+insert into category (name) values
+  ('BEST'),
+  ('GOOD'),
+  ('EXPERIMENTAL');
+
+
+create table map_version (
+  id serial primary key,
+  map_id integer not null references map(id),
+  version integer not null check (version > 0),
+  url varchar(256) not null check (url like 'http%'),
+  thumbnail varchar(256) not null check (thumbnail like 'http%'),
+  category_id integer not null references category(id),
+  description varchar(4096),
+  date_created timestamptz not null default now(),
+  date_updated timestamptz not null default now()
```

Comment:
Ok, I don't quite get what those timestamps are supposed to represent.
Doesn't the map table store the "last updated" date already?

Reply:
The first version of a map and the map date_created likely are going to always match. The map table is the umbrella for all of the different map versions that can be uploaded.

A new map version being uploaded gets a new row and hence a date_created. If we change the category, then date_updated would change.

Reply:
Ok, but wouldn't `map.date_created` and `map.date_updated` always be the "oldest"/newest entry of the corresponsing values in `map_version`?
I'm just thinking that there's some redundancy involved here

Reply:
`map.date_updated` would be updated if the name of a map is changed. It would not be updated if the `map_version` table is touched. `map.date_created` might very well correspond to the 'oldest' entry of `date_created` in `map_version`. I think at that point if you want to answer a query like "get me the list of maps by order of creation date", doing a window query on map_version is something of a pain. One would also want a created & updated date on `map` so that you can know if the name has been modified since creation.